tools/rpkg/R/print__duckdb_explain.R

```diff
@@ -0,0 +1,13 @@
+#' DuckDB EXPLAIN query tree
+#' @rdname duckdb_explain-class
+#' @export
+duckdb_explain <- setClass("duckdb_explain", contains = "data.frame", slots = list(explain_key = "character", explain_value = "character"))
+
+#' @rdname duckdb_explain-class
+#' @usage NULL
+#' @export
+print.duckdb_explain <- function(x, ...) {
+  if (!all(names(x) %in% c("explain_key", "explain_value"))) NextMethod("print")
+  if (nrow(x) > 0 & all(!is.na(x))) cat(paste0(x$explain_key, "\n", x$explain_value, collapse = ""))
```

Comment:
What's the purpose of `all_(!is.na(x))` here?

Reply:
There are two cases, when "zero" output occurs. Given that `x` is truly `NULL`, e.g. `x[FALSE,]`, that can be captured with `nrow(x)>0` and the out of "reasonable bounds" case such as `x[4,]` resulting in ugly printing of `NA`s  and can be captured using `all(!is.na(x))`. This escalates in situation like` x[4:50,]` and is better to silence.